crates/driver/src/domain/competition/solution/encoding.rs

```diff
@@ -182,92 +187,43 @@ pub fn tx(
                     &flashloan_wrapper.helper_contract,
                 ),
-            );
-
-            // Transfer tokens from flashloan wrapper to user (i.e. borrower) to later allow
-            // settlement contract to pull in all the necessary sell tokens from the user.
-            let tx = contracts::ERC20::at(
-                &contracts.settlement().raw_instance().web3(),
-                flashloan.token.into(),
-            )
-            .transfer_from(
-                flashloan_wrapper.helper_contract.address(),
-                flashloan.borrower.into(),
-                flashloan.amount.0,
-            )
-            .into_inner();
-            pre_interactions.insert(
-                1,
-                eth::Interaction {
-                    target: tx.to.unwrap().into(),
-                    value: eth::U256::zero().into(),
-                    call_data: tx.data.unwrap().0.into(),
-                },
-            );
-
-            // Since the order receiver is expected to be the setttlement contract, we need
-            // to transfer tokens from the settlement contract to the flashloan wrapper
-            let tx = contracts::ERC20::at(
-                &contracts.settlement().raw_instance().web3(),
-                flashloan.token.into(),
-            )
-            .transfer(
-                flashloan_wrapper.helper_contract.address(),
-                repayment_amount,
-            )
-            .into_inner();
-            post_interactions.push(eth::Interaction {
-                target: tx.to.unwrap().into(),
-                value: eth::U256::zero().into(),
-                call_data: tx.data.unwrap().0.into(),
-            });
-
-            // Allow flash loan lender to take tokens from wrapper contract
-            post_interactions.push(approve_flashloan(
-                flashloan.token,
-                repayment_amount.into(),
-                flashloan.lender,
-                &flashloan_wrapper.helper_contract,
-            ));
+                // Move the loaned money from the helper contract to the borrower address (most
+                // likely the user)
+                take_out_from_tracker(flashloan, flashloan_wrapper, flashloan_tracker),
+            ];
+            pre_interactions.splice(0..0, flash_loan_pre_interactions); // prepend interactions
+
+            let flash_loan_post_interactions = vec![
+                // Allow the flashloan lender to pull funds from the borrower contract (= pay back
+                // flash loan) after settlement execution
+                allow_lender_pull_funds(contracts, flashloan, flashloan_wrapper, repayment_amount),
+                // Call payBack() on the tracker; the user is expected to re-pay fee via post-hook
+                pay_back_via_tracker(flashloan, flashloan_wrapper, flashloan_tracker),
+            ];
+            post_interactions.splice(0..0, flash_loan_post_interactions); // prepend interactions
```

Comment:
I can't find the thread anymore, but wasn't this one supposed to happen at the end? 🤔 Not sure anymore.